### DISCUSSION TITLE
How to set up the contrast for multiple groups in `gseapy.gsea`?

### DISCUSSION THREAD

**Discussion by danli349** [zqfang/GSEApy#291] [General]
Hello:

If the `cls` file contains more than 2 groups, such as `['parental','parental','s40','s40','s42','s42']`, how should I set up the contrast when calling `gseapy.gsea` if I want to compare `parental_vs_s40`,  `parental_vs_s42` and `s42_vs_s40`? 

Thanks a lot

**Comment by zqfang** [comment]
I think the easiest way to do this is to subset the groups first, then run the gsea . It just two extra line of code  you need in python

```
df_groups = df.loc[,groups]
cls_vector = groups
gsea(...)
```

**Reply by danli349** [reply]
> I think the easiest way to do this is to subset the groups first, then run the gsea . It just two extra line of code you need in python
> 
> ```
> df_groups = df.loc[,groups]
> cls_vector = groups
> gsea(...)
> ```

That is great! Thanks
